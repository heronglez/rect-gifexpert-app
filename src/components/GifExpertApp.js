import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Euphoria'])

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                { 
                    categories.map(category => 
                        <GifGrid key={category} category={ category } />
                    )
                }
            </ol>
        </>
    )
}