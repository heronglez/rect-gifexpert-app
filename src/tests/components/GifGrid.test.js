import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => { 

    
    const category = 'Hola Mundo';

    test('debe de mostrarse correctamente', () => { 

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://localhost/xyz',
            title: 'cualquier cosa',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);

    })

 })