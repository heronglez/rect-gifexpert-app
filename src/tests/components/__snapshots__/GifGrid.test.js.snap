// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `ShallowWrapper {}`;
